### ISSUE TITLE
Page modal action: Undefined array key "mountedActionForm"

### ISSUE THREAD

**Issue by tanthammar** [laravel-filament/filament#1410]
### Package

filament/filament

### Package Version

v2.9.16

### Laravel Version

v8.81.0

### Livewire Version

v2.10.1

### PHP Version

v8.1.2

### Bug description

Trying to add a modal action button with confirmation on a custom admin Page.
No problem rendering the page, it is when I click the generated button I get an error:
```shell 
Undefined array key "mountedActionForm"
```

![image](https://user-images.githubusercontent.com/21239634/151954773-3c253e3f-93e4-4342-8bf3-59cd3064cea0.png)

### Steps to reproduce

Page:
```php
<?php

namespace App\Filament\Pages;

use Filament\Forms\ComponentContainer;
use Filament\Pages\Actions\ButtonAction;
use Filament\Pages\Page;
use Illuminate\Support\Facades\Auth;
use Filament\Forms\Components\TextInput;

/**
 * @property ComponentContainer|mixed|null $passwordForm
 * @property ComponentContainer|mixed|null $userNameEmailForm
 */
class Profile extends Page
{
    protected static ?string $navigationIcon = 'heroicon-o-user';
    protected static ?string $navigationGroup = 'Account';
    protected static string $view = 'user.profile';
    public string $password = '';
    public string $deleteMsg = '';

    protected function getForms(): array
    {
        return [
            'userNameEmailForm' => $this->makeForm()
                ->schema([/* name email fields */])
                ->model(Auth::user()),
            'passwordForm' => $this->makeForm()
                ->schema([/* psw fields */])
                ->model(Auth::user()),
        ];
    }

    protected function getActions(): array
    {
        return [
            ButtonAction::make(__('Delete account'))
                ->action('deleteUser')
                ->form([
                        TextInput::make($password)
                            ->label('Password')
                            ->password()
                            ->required()
                            ->currentPassword()
                            ->autocomplete('off')
                ])
                ->color('danger')
                ->icon('heroicon-o-key')
                ->requiresConfirmation(),
        ];
    }

    private function deleteUser()
    {
        //return route()
    }
}
```

view
```blade
<x-filament::page>
    @if($deleteMsg)
        <div class="rounded bg-danger text-white p-6">
            <p>{{ $deleteMsg }}</p>
        </div>
    @endif
    <form>
        {{ $this->userNameEmailForm }}
        <!-- submit btn -->
    </form>
    <form>
        {{ $this->passwordForm }}
       <!-- submit btn -->
    </form>
</x-filament::page>
```

### Relevant log output

Flare exception public link: [https://flareapp.io/share/OmVKzlRP#F60](https://flareapp.io/share/OmVKzlRP#F60)

**Comment by danharrin** [comment]
Hey, you need to make sure that you merge in the page's forms as well as your own in `getForms()`:

```php
protected function getForms(): array
{
    return array_merge(parent::getForms(), [
        'userNameEmailForm' => $this->makeForm()
            ->schema([/* name email fields */])
            ->model(Auth::user()),
        'passwordForm' => $this->makeForm()
            ->schema([/* psw fields */])
            ->model(Auth::user()),
     ]);
}
```

**Comment by romaldyminaya** [comment]
@tanthammar I have the same problem,

Could you please share how did you solve this ?

**Comment by romaldyminaya** [comment]
As I figured out what problem this has, I'll do a pull request with the solution.

**Comment by tanthammar** [comment]
There was nothing wrong with Filament. I forgot to do `array_merge(parent::getForms(),...` Like Dan commented above.

**Comment by danharrin** [comment]
Fixed in #1773, no more `array_merge()`!
